blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/BluefloodCounterRollupTest.java

```diff
@@ -1,93 +1,758 @@
+        //          | <-- one time unit --> |                       |
+        //    ------+-----------------------+-----------------------+-----
+        //          1                       1                       1
+        //          2                       2                       2
+        //          3                       3                       3
+        //          4                       5                       6
+        //          L                       L                       L
+        //          |                                               |
+        //          | <-------------- two time units -------------> |
+        //                             (3+4+5)/2=6
+        //
+        //
+        // Samples As Intervals:
+        //                                                           _______________________
+        //                                   _______________________|          5            |
+        //           _______________________|          4            |                       |
+        //          |          3            |                       |                       |
+        //          |                       |                       |                       |
+        //          |                       |                       |                       |
+        //    ------+-----------------------+-----------------------+-----------------------+-----
+        //          1                       1                       1                       1
+        //          2                       2                       2                       2
+        //          3                       3                       3                       3
+        //          4                       5                       6                       7
+        //          L                       L                       L                       L
+        //          |                                                                       |
+        //          | <------------------------- three time units ------------------------> |
+        //                                         (3+4+5)/3=4
+        //
+
+        // given
+        Points<SimpleNumber> input = new Points<SimpleNumber>();
+        input.add(new Points.Point<SimpleNumber>(1234L, new SimpleNumber(3L)));
+        input.add(new Points.Point<SimpleNumber>(1235L, new SimpleNumber(4L)));
+        input.add(new Points.Point<SimpleNumber>(1236L, new SimpleNumber(5L)));
+        // when
+        BluefloodCounterRollup rollup = BluefloodCounterRollup.buildRollupFromRawSamples(input);
+        // then
+        Number count = rollup.getCount();
+        assertNotNull(count);
+        assertThat(count, is(CoreMatchers.<Number>instanceOf(Long.class)));
+        assertEquals(12L, count.longValue());
+        assertEquals(6.0d, rollup.getRate(), 0.00001d); // (3+4+5)/(3-1)=6
+        assertEquals(3, rollup.getSampleCount());
```

Comment:
Ok rethinking this you're right, there's an alternate way of looking at rate that the current implementation is correct for.

If we were to capture the speed of walking at time 1, 2, and 3, then the average speed for the entire interval is val(1) + val(2) + val(3) / 3 - 1.  So the rate here means the average speed over the interval, which is possibly how the original devs want to treat "rate" as for metrics.